scripts/build-docker-image.sh

```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/env bash
+
+
+set -euxo pipefail
+
+ARCH=${ARCH:-$(uname -m)}
+
+# Set the correct build target and update the arch if required.
+if [[ "$ARCH" = "amd64" ]]; then
+  BUILD_TARGET="x86_64-unknown-linux-gnu"
+elif [[ "$ARCH" = "arm64" ]]; then
+  BUILD_TARGET="aarch64-unknown-linux-gnu"
+elif [[ "$ARCH" = "aarch64" ]]; then
+  BUILD_TARGET="aarch64-unknown-linux-gnu"
+  ARCH="arm64"
+fi
+
+# Images to use and build.
+IMG_DEPS=${IMG_DEPS:-"ghcr.io/getsentry/relay-deps:$ARCH"}
+IMG_VERSIONED=${IMG_VERSIONED:-"relay:latest"}
+
+# Build a builder image with all the depdendencies.
+args=(--progress auto)
+if docker pull -q "$IMG_DEPS"; then
+  args+=(--cache-from "$IMG_DEPS")
+fi
+
+docker buildx build \
+    "${args[@]}" \
+    --cache-to type=inline \
+    --platform "linux/$ARCH" \
+    --tag "$IMG_DEPS" \
+    --target relay-deps \
+    --file Dockerfile.builder \
+    .
+
+# Build the binary inside of the builder image.
+docker run \
+    -v "$(pwd):/work" \
```

Comment:
`$(pwd)` => `$PWD` (var lookup is free compared to a subshell and subprocess)

I also like to be explicit and use `--volume` and state whether it's `:ro` or `:rw` (I don't even remember the default)